core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingPublisher.java

```diff
@@ -170,37 +195,59 @@ private void completeCurrentBodyAndCreateNewIfNeeded(ByteBuffer byteBuffer) {
 
             if (shouldCreateNewDownstreamRequestBody) {
                 long chunkSize = calculateChunkSize(dataRemaining);
-                currentBody = initializeNextDownstreamBody(upstreamSize != null, chunkSize, currentChunk);
+                currentBody = initializeNextDownstreamBody(upstreamSize != null, chunkSize, nextChunk);
             }
         }
 
         private int amountRemainingInChunk() {
-            return Math.toIntExact(currentBody.maxLength - currentBody.transferredLength);
+            return Math.toIntExact(currentBody.maxLength() - currentBody.receivedBytesLength());
         }
 
+
         private void completeCurrentBody() {
-            log.debug(() -> "completeCurrentBody for chunk " + chunkNumber.get());
+            log.debug(() -> "completeCurrentBody for part " + currentBody.partNumber());
+            // For unknown content length, we always create a new DownstreamBody because we don't know if there is data
+            // left or not, so we need to only send the body if there is actually data
+            long bufferedLength = currentBody.receivedBytesLength();
+            if (bufferedLength == 0) {
```

Comment:
If possible just in response to this comment can you tell me how we will end up `bufferedLength == 0` here  ? As in for touch files () byte files)